### DISCUSSION TITLE
Can I know where SILE's named colors come from?

### DISCUSSION THREAD

**Discussion by ctrlcctrlv** [sile-typesetter/sile#1271] [Q&A] [answered]
https://github.com/sile-typesetter/sile/blob/985b9c3188ef9ee55046215aa2a94cf4f96d35f7/core/colorparser.lua#L1

Also, should a standardized set from a pacakge be used instead?

**Comment by Omikhleia** [comment]
The **color** package documentation says it is "as in HTML", and this actually seems to be the same as https://www.w3.org/TR/css-color-3/#svg-colorhttps://www.w3.org/TR/css-color-3/#svg-color

That is, the 147 or so - I didn't count them - so-called "X11colors" that almost all browsers implemented, and that the SVG and CSS standard ended up standardizing or at least acknowledging.

If true, this is a fairly reasonable assumption. What would be the benefit of an extra package?

EDIT: The (currently 2021) Draft for CSS4 also has them, and much longer descriptions of alternate color profiles https://www.w3.org/TR/css-color-4/#named-color - Colors are such a mess... As for supporting other color profiles, such as Pantone, see e.g. for inspiration https://tex.stackexchange.com/questions/202175/using-pantone-spot-colors-in-xelatex

**Comment by simoncozens** [comment, marked as answer]
Yes, these are the standard CSS colors plus rebeccapurple.

**Comment by alerque** [comment]
I typically use a colorspace manipulation library such as [colors](https://luarocks.org/modules/luarocks/colors) in my own projects. There are several that have pros and cons for various use cases. I don't think we need to require any of them as dependencies, but it might be useful to have shim packages that automatically extend SILE color handling with features from those libraries if any of them are present/called for.

**Reply by Omikhleia** [reply]
@ctrlcctrlv Did you get the expected answer?

**Comment by Omikhleia** [comment]
I picked an answer.
I don't see the point of having questions remaining unanswered for so long when several answers were provided, and it does not lead to any enhancement issue.
